### PR TITLE
Network refactor

### PR DESCRIPTION
**Pull request by Globalika**
decoupled network in separate class
added apollo protocol
added error handling

### CODE REVIEW

**Review by khrystyna-sb — APPROVED** [review]

File: CountriesOfTheWorld/CountriesOfTheWorld/Apollo.swift [hunk 1]
```diff
@@ -8,18 +8,52 @@
 import Apollo
 import Foundation
 
-class Apollo {
+protocol ApolloClientProtocol {
+    func getCountries(completion: @escaping (Result<[CountriesQuery.Data.Country], Error>) -> Void)
```
Comment (ambrusha) [inline CountriesOfTheWorld/CountriesOfTheWorld/Apollo.swift:12]: in line 10: `public typealias Countries = CountriesQuery.Data.Country` and so on...

File: CountriesOfTheWorld/CountriesOfTheWorld/SceneDelegate.swift [hunk 2]
```diff
@@ -22,10 +22,10 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         window = UIWindow(frame: windowScene.coordinateSpace.bounds)
         window?.windowScene = windowScene
 
-        let mainController = CountriesController()
+        let mainController = CountriesController(NetworkManager(client: Apollo()))
         let mainNC = UINavigationController(rootViewController: mainController)
 
-        let detailController = CountryDetailsController()
+        let detailController = CountryDetailsController(NetworkManager(client: Apollo()))
```
Comment (ambrusha) [inline CountriesOfTheWorld/CountriesOfTheWorld/SceneDelegate.swift:28]: do you need to have TWO `NetworkManager`s ?

Comment (Globalika) [reply CountriesOfTheWorld/CountriesOfTheWorld/SceneDelegate.swift:28]: my network manager can have(will have i believe) additional functionality like getting flags

File: CountriesOfTheWorld/CountriesOfTheWorld/Apollo.swift [hunk 3]
```diff
@@ -8,18 +8,55 @@
...
 
-class Apollo {
+typealias CountryLite = CountriesQuery.Data.Country
+typealias CountryDetailed = CountryQuery.Data.Country
 
-    static let shared = Apollo()
+protocol ApolloClientProtocol {
+    func getCountries(completion: @escaping (Result<[CountryLite], Error>) -> Void)
+    func getCountry(code: String,
+                    completion: @escaping (Result<CountryDetailed, Error>) -> Void)
```
Comment (ambrusha) [inline CountriesOfTheWorld/CountriesOfTheWorld/Apollo.swift:17]: nit: use one line for this. Readability is ok.

File: CountriesOfTheWorld/CountriesOfTheWorld/SceneDelegate.swift (hunk 2 above)
Comment (ambrusha) [reply CountriesOfTheWorld/CountriesOfTheWorld/SceneDelegate.swift:28]: I mean next:
Can you just in line 24 create object:
` let networkManager = NetworkManager(client: Apollo())` and then pass it to `mainController` and `detailController`?

**Review by ambrusha — APPROVED** [review]

**Pull request merged**